source/MRMesh/MRCone3.h

```diff
@@ -0,0 +1,65 @@
+#pragma once
+#include "MRLine.h"
+
+namespace MR
+{
+
+// Base class for cone parameterization
+
+template <typename T>
+class Cone3
+{
+public:
+    Cone3()
+    {}
+
+    // inAxis -- apex position and main axis direction.  For any cone point dot( point , direction ) >=0
+    // inAngle -- cone angle, main axis vs side
+    // inHeight -- cone inHeight
+    // main axis direction could be non normalized.
+    Cone3( const Line3<T>& inAxis, T inAngle, T inHeight )
+        :
+        axis( inAxis ),
+        angle( inAngle ),
+        height( inHeight )
+    {}
+
+    MR::Vector3<T>& center( void )
```

Comment:
no need in MR:: prefixes inside the namespace

Reply:
ok, but I keep explisit MR:: in ConeApproximation.h  file. 
There Eigen and MR Vectors works together, and I think in that particular file better use explisit naming.